components/nimbus/src/enrollment.rs

```diff
@@ -17,7 +15,18 @@ use std::{
     time::{Duration, SystemTime, UNIX_EPOCH},
 };
 
-const DB_KEY_GLOBAL_USER_PARTICIPATION: &str = "user-opt-in";
+cfg_if::cfg_if! {
+    if #[cfg(feature = "stateful")] {
+        use crate::{
+            persistence::{Database, Readable, StoreId, Writer},
+            EnrolledExperiment,
+        };
+
+        const DB_KEY_GLOBAL_USER_PARTICIPATION: &str = "user-opt-in";
+    }
+}
+
+#[cfg_attr(not(feature = "stateful"), allow(unused))]
```

Comment:
Massive thank you @yashikakhurana for finding `cfg_attr`, this is a great adjustment 🎉